compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt

```diff
@@ -55,6 +58,8 @@ inline val FirMemberDeclaration.isFromEnumClass: Boolean get() = status.isFromEn
 
 inline val FirPropertyAccessor.modality get() = status.modality
 inline val FirPropertyAccessor.visibility get() = status.visibility
+inline val FirPropertyAccessor.allowsToHaveFakeOverride: Boolean
+    get() = !Visibilities.isPrivate(visibility) && this != Visibilities.INVISIBLE_FAKE
```

Comment:
Counterpart I referred is: https://github.com/JetBrains/kotlin/blob/master/compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/PropertyGenerator.kt#L180

Technically, `INVISIBLE_FAKE` is not used in `fir` so far, but that visibility is self-explanatory, and having it here won't hurt...

Reply:
I think you mean `get() = !Visibilities.isPrivate(visibility) && visibility != Visibilities.INVISIBLE_FAKE` (`visibility` instead of `this`). Same for the code above.

Reply:
Duh, you're right! Thanks for spotting this. Will be fixed with https://github.com/JetBrains/kotlin/pull/3259